backend/api/simulation.py

```
@@ -1726,15 +2010,105 @@ def _safe_scene_id():
                         # Update orchestrator state to match database
                         if next_scene_id:
                             # Find the scene index for the new scene
+                            next_scene_obj = None
                             for i, scene in enumerate(orchestrator.scenes):
                                 if scene.get('id') == next_scene_id:
                                     orchestrator.state.current_scene_index = i
+                                    next_scene_obj = scene
                                     break
                             orchestrator.state.turn_count = 0
                             print(f"[DEBUG] TURN COUNT RESET TO 0 ON GOAL VALIDATION PROGRESSION")
```

Ruff (0.13.3)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)